domains/client/relayer/src/lib.rs

```diff
@@ -1,21 +1,22 @@
 #![warn(rust_2018_idioms)]
+#![deny(unused_crate_dependencies)]
```

Comment:
I'll have to remove this line. It results in following confusing error:
```
error: external crate `alloc` unused in `domain_client_message_relayer`: remove the dependency or add `use alloc as _;`
  |
note: the lint level is defined here
 --> domains/client/relayer/src/lib.rs:2:9
  |
2 | #![deny(unused_crate_dependencies)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^

error: external crate `compiler_builtins` unused in `domain_client_message_relayer`: remove the dependency or add `use compiler_builtins as _;`

error: external crate `panic_unwind` unused in `domain_client_message_relayer`: remove the dependency or add `use panic_unwind as _;`

error: external crate `proc_macro` unused in `domain_client_message_relayer`: remove the dependency or add `use proc_macro as _;`

error: external crate `test` unused in `domain_client_message_relayer`: remove the dependency or add `use test as _;`

error: could not compile `domain-client-message-relayer` (lib test) due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
error: could not compile `domain-client-message-relayer` (lib) due to 5 previous errors
```

To reproduce `-Z build-std` is needed, for example:
```bash
cd domains/client/relayer
cargo clippy --all-targets -Z build-std --target x86_64-unknown-linux-gnu
```

I'm not sure where it comes from because there is no such things in the crate and I don't think any of the very few macros generate it either (I checked macro expansion). Might be compiler issue: https://github.com/rust-lang/rust/issues/122105